infra/cifuzz/config_utils.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for getting the configuration CIFuzz needs to run."""
+
+import logging
+import enum
+import os
+import json
+
+
+class Config:  # pylint: disable=too-few-public-methods,too-many-instance-attributes
+  """Object containing constant configuration for CIFuzz."""
+
+  class Platform(enum.Enum):
+    """Enum representing the different platforms CIFuzz runs on."""
+    EXTERNAL_GITHUB = 0  # Non-OSS-Fuzz on GitHub actions.
+    INTERNAL_GITHUB = 1  # OSS-Fuzz on GitHub actions.
+    INTERNAL_GENERIC_CI = 2  # OSS-Fuzz on any CI.
+
+  def __init__(self):
+    """Get the configuration from CIFuzz from the environment. These variables
+    are set by GitHub or the user."""
+    self.project_name = os.getenv('OSS_FUZZ_PROJECT_NAME')
+    self.project_repo_name = os.path.basename(os.getenv('GITHUB_REPOSITORY'))
```

Comment:
nit: i realise this is only a draft, but I would suggest instead of doing this to break this up into subclasses rather than trying to make a one-size-fits-all thing that works on all CIs. 

i.e. Config -> GitHubConfig, OssFuzzGitHubConfig, GenericCIConfig

More generally, we should start thinking about how we support more CIs more cleanly. e.g. ideally all CI support can all be encapsulated into an interface and be self contained in a single file/dir.

e.g.

```python
# ci_provider.py
class CIProvider:
  def get_config():
    ...
  
  def get_artifacts_storage():
    ...
 
  def get_builder():
    ...

  # whatever else we need and can abstract away
```


```python
# github/provider.py
class GitHubCIProvider(CIProider):
  ...
```

etc etc. 

Rather than lots of:

```python
if github: 
elif oss_fuzz_github:
else:
```

and implementations for logic based on them scattered throughout the core CIFuzz code.

Reply:
Done.